src/main/java/com/puppycrawl/tools/checkstyle/DetailAstImpl.java

```diff
@@ -166,7 +166,6 @@ public void addChild(DetailAST child) {
         if (child != null) {
             final DetailAstImpl astImpl = (DetailAstImpl) child;
             astImpl.setParent(this);
-            astImpl.previousSibling = (DetailAstImpl) getLastChild();
```

Comment:
1. If this condidition is true (which means this is the very first child)
```
        if (temp == null) {
            firstChild = (DetailAstImpl) child;
        }
```
then the `previousSibling` won't be set because you removed it, but before your change it would have been set.

2. `astImpl.previousSibling = (DetailAstImpl) getLastChild();` here we set it to the last child, but in `((DetailAstImpl) temp).setNextSibling(child);` we set to `parent` or `this` which doen't look like last child.

Reply:
1) 
If the temp is null then the node has no children something like
```
GrandParent
    |
    |
  Parent
```
and we call `parent.addChild(child)` In this case temp will become null and if the node doesn't have any children then There should be no issue with PreviousSibiling because it is always going to be null.
2)

Our here temp would be our last child means 
```
 Parent
    |
    |
  child1---child2
```
this is the current tree and lets suppose we call `parent.addChild(child3)` In this case our temp would be equal to child2 and know child2 is calling its method `setNextSibling` that means `this` would be equal to `child2` in `setNextSibiling` and in `this.nextSibling = (DetailAstImpl) nextSibling;` this statement in setNextSibiling Method will assign the param node nextSibiling as a next node of child2 and `((DetailAstImpl) nextSibling).previousSibling = this;` at last nextsibiling means `child3` would be assigned to this means `child2`